src/cargo/util/toml/mod.rs

```diff
@@ -895,7 +896,7 @@ impl TomlDependency {
                 let loc = git.to_url()?;
                 SourceId::for_git(&loc, reference)?
             },
-            (None, Some(path)) => {
+            (None, Some(path), _) => {
```

Comment:
Similar to above, I think we should return an error if the registry is specified but a path is also specified